### PR TITLE
Announce 4.139

### PR DESCRIPTION
**Pull request by fredemmott**

### CODE REVIEW

**Comment by fredemmott** [comment]
Also SimpleParser JSON BC breaks (thanks @lexidor :) )

File: _posts/2021-12-09-hhvm-4.139.markdown [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- The IDE quickfix for missing methods now creates `async` stub methods if the return type is `Awaitable`.
+- Added IDE quickfix for int, float, or string constants that are declared without a type.
+- `json_decode()` is now faster, and consumes less memory.
+
+# Breaking Changes
+
+- It is now an error to read static properties in functions that do not have the `globals` or `read_globals` capabilties.
+- It is now an error to write static properties in functions that do not have the `globals` capabilty.
+- `json_decode()` is now stricter about the input it accepts (per JSON/ECMA-404):
+  - leading 0s are no longer permitted for any numbers.
```
Comment (lexidor) [inline _posts/2021-12-09-hhvm-4.139.markdown:27]: `01` and `-01` still parse as `1` and `-1` respectively. `[01]` and `[-01]` are now a correct syntax error 🎉.
https://github.com/facebook/hhvm/issues/8797#issuecomment-990146598 contains the test results.

Comment (fredemmott) [reply _posts/2021-12-09-hhvm-4.139.markdown:27]: also `{"foo": 01}` is invalid

Comment (fredemmott) [reply _posts/2021-12-09-hhvm-4.139.markdown:27]: @Matzeb, what's the intended behavior of https://github.com/facebook/hhvm/commit/d45c940a9aec162a81d2c812eb644148fa97a758 ?

```
hphpd> =json_decode('03')
3
hphpd> =json_decode('-03')
-3
hphpd> =json_decode('{"foo": 03}')
null
hphpd> =json_decode('{"foo": -03}')
null
hphpd> =json_decode('{"foo": 3}')
stdClass Object
(
    [foo] => 3
)

hphpd> =json_decode('{"foo": -3}')
stdClass Object
(
    [foo] => -3
)
```

Comment (lexidor) [reply _posts/2021-12-09-hhvm-4.139.markdown:27]: Don't forget to toggle the `bool $assoc` argument as well. The parser accepts more invalid inputs when passing the non-default value `true`.

Comment (fredemmott) [reply _posts/2021-12-09-hhvm-4.139.markdown:27]: @lexidor are you seeing any observable behavior change with leading 0s? Comparing to 4.135, I'm not seeing one

Comment (fredemmott) [reply _posts/2021-12-09-hhvm-4.139.markdown:27]: Yep, `json_decode('{"foo": -03}', true)` is one

Comment (lexidor) [reply _posts/2021-12-09-hhvm-4.139.markdown:27]: I was comparing against 4.88. That was when I wrote the original issue.

**Pull request merged**